main.js

```diff
@@ -11,28 +11,55 @@ function generateNumbers() {
     return allNumbers;
 }
 
-function displayNumbers(allNumbers) {
-    const numbersOnScreen = allNumbers.length;
+function displayNumbers(randomNumbers) {
+    const sortedNumbers = randomNumbers.slice(0).sort((a, b) => a - b);
+    const numbersOnScreen = randomNumbers.length;
+    let selectedOrder = [];
 
-     // using a for loop instead of forEach as the array.pop changes the length over each iteration
-     for (let i = 0; i < numbersOnScreen; i++) {
-        let randNumber = allNumbers.pop();
+    // using a for loop instead of forEach as the array.pop changes the length over each iteration
+    for (let i = 0; i < numbersOnScreen; i++) {
+        let randNumber = randomNumbers.pop();
         let number = document.createElement('div');
 
         number.textContent = randNumber;
         number.setAttribute('class', 'numberBox');
+
         document.querySelector('#play_area').appendChild(number);
     }
+
+    document.querySelectorAll('.numberBox').forEach((numberBox) => {
+        numberBox.addEventListener('click', () => {
+            const { textContent, classList } = numberBox;
+            const numberBoxValue = parseInt(textContent);
+
+            // check if already clicked
+            if (!classList.contains('clicked')) {
```

Comment:
i still dont like this, can you put the unclear bits of the code into semantic functions? like this if, and the if below, and the length if near the bottom.